internal/scylla/client.go

```diff
@@ -280,3 +327,30 @@ func (c *Client) findAndSaveAccountID(ctx context.Context) error {
 
 	return nil
 }
+
+// cloneRequest returns a clone of the provided *http.Request to make it retryable
+func cloneRequest(req *http.Request) (cpy *http.Request, err error) {
+	cpy = req.Clone(req.Context())
+	if req.Body == nil || req.Body == http.NoBody {
+		return req, nil
+	}
+
+	if req.GetBody != nil {
+		cpy.Body, err = req.GetBody()
+		if err != nil {
+			return nil, err
+		}
+		return cpy, nil
+	}
```

Comment:
How about we do something more resilient? If the req.GetBody is nil (which should not be the case, since we are using http.NewRequest which always is setting the GetBody), then we could do something like:

```go
	if req.GetBody == nil {
		var (
			buf bytes.Buffer
			lr  = &io.LimitedReader{
				R: req.Body,
				N: maxBody,
			}
		)

		body = io.TeeReader(lr, &buf)

		req.GetBody = func() (io.ReadCloser, error) {
			if lr.N <= 0 {
				return nil, shared.ErrBodyTooLarge
			}

			return io.NopCloser(bytes.NewReader(buf.Bytes())), nil
		}
	}

	// ...

	n, err := io.Copy(&buf, body)
	if n == 0 {
		return errors.New("httpx: error signing request body: body is empty")
	}
	if err != nil {
		return errors.New("httpx: error verifying request body: %w", err)
	}

	// ...

	req.Body, err = req.GetBody()
	if err != nil {
		return errors.New("httpx: error rewinding request body: %w", err)
	}
```

Reply:
selected the wrong Rafael?

Reply:
> could you please clarify if sending only 1MB of original data on retry is expected behavior.

@dkropachev Sure thing, let me explain - usually when reading req.Body we would want to put limit on the amount of memory the request could take, to prevent potential dos, which part of the times happen due to simple error. Reading req.Body with a Limiter is especially important on the server-side code, which we do.

For client code is also important here, as we take input from user. Since we are not validating request payloads against an OpenAPI schema, the least we can do is to put a reasonable limit on the payload body - and 1MiB seems enough. The prevent cases where user would erroneously put in their template e.g. `node_type = ${file("${path.module}/large_file.tar.gz")}` and then tried to execute.

We also should limit the body on the initial request, not only the retry. It can be done in a separate PR though, it's not a requirement for this PR.

Reply:
@rjeczalik , so, wouldn't it better to address this issue on separate PR for initial request, which will automatically apply this fix for retry body ?